packages/ai-jsx/src/stream/index.ts

```diff
@@ -56,14 +71,38 @@ async function* renderToJsonEvents(renderable: Renderable): AsyncGenerator<Strea
     }
   }
 
-  return { type: 'complete' } as StreamEvent;
+  return { type: 'complete' };
+}
+
+function jsonReplacerFromSerializer(serializer: ElementSerializer) {
+  return (key: string, value: unknown) => {
+    let currentValue: unknown = value;
+
+    if (currentValue !== null && typeof currentValue === 'object') {
+      while (isIndirectNode(currentValue)) {
+        currentValue = getReferencedNode(currentValue);
+      }
+
+      if (isElement(currentValue)) {
+        return serializer(currentValue);
+      }
+    }
+
+    return currentValue;
+  };
+}
+
+function jsonReviverFromDeserializer<T>(deserializer: ElementDeserializer<T>) {
+  return (key: string, value: Jsonifiable) => deserializer(value) || value;
```

Comment:
```suggestion
  return (key: string, value: Jsonifiable) => deserializer(value);
```

Maybe this can be removed?

Reply:
I left this way for symmetry with `ElementSerializer` and for clarity of types -- `ElementDeserializer` is responsible for deserializing anything that was serialized by `ElementSerializer` but otherwise returns `undefined` to let things deserialize as-is. (Namely `StreamEvent` needs to be preserved.) I started to change the semantics but realized that it was going to nerf the type signature.